src/Parser.php

```diff
@@ -527,9 +550,7 @@ protected function parsePathQueryAndFragment(string $uri): array
     protected function fallbackParser(string $uri): array
     {
         //1 - we split the URI on the first detected colon character
-        $parts = explode(':', $uri, 2);
-        $remaining_uri = array_pop($parts);
-        $scheme = array_shift($parts);
+        list($scheme, $remaining_uri) = explode(':', $uri, 2) + [1 => null];
```

Comment:
This is wrong, because it assigns `$scheme` now if there's no colon instead of `$remaining_uri`.

Reply:
@trowski could you revert this change please ?

Reply:
maybe use the other construct
```php
$parts = explode(':', $uri, 2);
$remaining_uri = $parts[1] ?? $parts[0];
$scheme = isset($parts[1]) ? $parts[0] : null;
```